tethysext/workflows/results/image_workflow_result.py

```diff
@@ -0,0 +1,158 @@
+    buf.seek(0)
+    string = base64.b64encode(buf.read())
+    uri = urllib.parse.quote(string)
+    ```
+    """
+    CONTROLLER = 'tethysext.workflows.controllers.workflows.results_views.image_workflow_results_view.ImageWorkflowResultView'  # noqa: E501
+    TYPE = 'image_workflow_result'
+
+    __mapper_args__ = {
+        'polymorphic_on': 'type',
+        'polymorphic_identity': TYPE
+    }
+
+    def __init__(self, *args, **kwargs):
+        """
+        Constructor.
+
+        Args:
+        """
+        super().__init__(*args, **kwargs)
+
+    @property
+    def default_options(self):
+        """
+        Returns default options dictionary for the object.
+        """
+
+        default_options = super().default_options
+        default_options.update({
+            'no_dataset_message': 'No dataset found.'
+        })
+        return default_options
+
+    @property
+    def datasets(self):
+        if 'datasets' not in self.data:
+            self.data['datasets'] = []
+        return copy.deepcopy(self.data['datasets'])
+
+    @datasets.setter
+    def datasets(self, value):
+        data = copy.deepcopy(self.data)
+        data['datasets'] = value
+        self.data = data
+
+    @property
+    def image(self):
+        if 'image' not in self.data:
+            self.data['image'] = ''
+        return copy.deepcopy(self.data['image'])
+
+    @image.setter
+    def image(self, value):
+        data = copy.deepcopy(self.data)
+        data['image'] = value
+        self.data = data
+
+    def reset(self):
+        self.datasets = []
+
+    def _add_dataset(self, dataset):
+        """
+        Adds the dataset to the datasets array.
+
+        Args:
+            dataset(dict): The data.
+        """
+        datasets = self.datasets
+        datasets.append(dataset)
+        self.datasets = datasets
+
+    def _add_image(self, image_object):
+        """
+        Update image object.
+
+        Args:
+            image_object(dict): The image.
+        """
+        self.image = image_object
+
+    def image_from_matplotlib_figure(self, figure):
+        """
+        Adds a matplotlib plot figure object to the result.
+
+        Args:
+            figure(obj): matplotlib figure.
+        """
+        if not figure:
+            raise ValueError('The matplotlib plot figure must not be empty.')
+
+        buf = io.BytesIO()
+        figure.savefig(buf, format='png')
+        buf.seek(0)
+        string = base64.b64encode(buf.read())
+        uri = urllib.parse.quote(string)
+
+        d = {
+            'dataset': uri,
+        }
+        self._add_dataset(d)
+
+    def add_image(self, image, description=''):
+        """
+        Adds a image to the result.
+
+        Args:
+            image(str): image uri.  Add only one image.
+            description(str): description of the image.  Defaults to ''.
+        """
+        image_object = {
+            'image_object': image,
+            'image_description': description,
+        }
```

Comment:
Is there a reason we can't put the URI in the image object too and get rid of dataset?
```suggestion
        image_object = {
            'image_object': image,
            'image_description': description,
            'uri': uri,
        }
```